index.js

```diff
@@ -470,21 +494,45 @@ const createFocusTrap = function (elements, userOptions) {
         return this;
       }
 
-      updateTabbableNodes();
-
       state.active = true;
       state.paused = false;
       state.nodeFocusedBeforeActivation = doc.activeElement;
 
-      const onActivate =
-        activateOptions && activateOptions.onActivate
-          ? activateOptions.onActivate
-          : config.onActivate;
-      if (onActivate) {
-        onActivate();
+      const getOption = (optionName) => {
+        return activateOptions && activateOptions[optionName] !== undefined
+          ? activateOptions[optionName]
+          : config[optionName];
+      };
```

Comment:
This would be useful to other functions; let's promote this to the outer scope instead of keeping it in `activate()` only. No need to use it elsewhere in this PR, but `deactivate()` could make use of it, for example.

Reply:
I tried using it in `deactivate()` but it broke the return focus to origin functionality.

`deactivateOptions` has `returnFocus` but `config` has `returnFocusOnDeactivate`.

That's why I made this commit
https://github.com/focus-trap/focus-trap/pull/407/commits/d83bddee8fb73dd6fb85f8440580a7268ab504df

I can take another look at the function and see how I can make it more reusable.

Reply:
I see, in that case, don't worry about it. This is very minor.